pmd-core/src/main/java/net/sourceforge/pmd/PMD.java

```diff
@@ -470,14 +479,19 @@ public static int run(String[] args) {
         } finally {
             logHandlerManager.close();
             LOG.setLevel(oldLogLevel);
+            
             if (params.isBenchmark()) {
-                long end = System.nanoTime();
-                Benchmarker.mark(Benchmark.TotalPMD, end - start, 0);
+                final TimingReport timingReport = TimeTracker.stopGlobalTracking();
 
                 // TODO get specified report format from config
-                TextReport report = new TextReport();
-
-                report.generate(Benchmarker.values(), System.err);
+                final TimingReportRenderer renderer = new TextTimingReportRenderer();
+                try {
+                    // Don't close this writer, we don't want to close stderr
+                    final Writer writer = new OutputStreamWriter(System.err);
+                    renderer.render(timingReport, writer);
+                } catch (final IOException e) {
+                    System.err.println(e.getMessage());
```

Comment:
Shouldn't this use a Logger?

Reply:
I thought so, but decided against it. This is within the `main` for PMD's CLI. Outputting data back to the user of stdout / stderr is expected, and using a logger would make the result dependent of the logger's setup (logging to a file or not, log level, etc.). The `catch` block just above this `finally` is actually doing the very same thing, so it seems consistent.